### PR TITLE
Added creator field to discovery service

### PR DESCRIPTION
**Pull request by Markus92**
## 🚀 Pull Request Checklist

- **Title:**

This PR exposes the CKAN creators field in the discovery service.

- **Description:**

See title.

- **Context:**

The field creator is currently not exposed in the discovery service, but its an essential part of DCAT-AP. It follows the same structure as publisher.

- **Changes:**

- OpenAPI defintion updated
- PackageMapper updated
- Tests updated

- **Testing:**

- Changes have been tested using Postman and internal fork of userportal frontend

- **Screenshots (if applicable):**

N/A

- **Additional Information:**

Code mostly done by my colleague @kburger and gratefully borrowed from his branch.

- **Checklist:**
  - `[X]` I have checked that my code adheres to the project's style guidelines and that my code is well-commented.
  - `[X]` I have performed self-review of my own code and corrected any misspellings.
  - `[X]` I have made corresponding changes to the documentation (if applicable).
  - `[X]` My changes generate no new warnings or errors.
  - `[X]` I have added tests that prove my fix is effective or that my feature works.
  - `[X]` New and existing unit tests pass locally with my changes.

## Summary by Sourcery

Expose the 'creator' field in the discovery service by updating the OpenAPI definition and PackageMapper. Add corresponding tests to ensure proper functionality.

New Features:
- Expose the 'creator' field in the discovery service, following the same structure as the 'publisher' field.

Enhancements:
- Update OpenAPI definition to include the 'creator' field.
- Update PackageMapper to map the 'creator' field from CKAN packages.

Tests:
- Add tests to verify the correct parsing and mapping of the 'creator' field in CKAN packages.

### CODE REVIEW

File: src/main/java/io/github/genomicdatainfrastructure/discovery/utils/PackageShowMapper.java [hunk 1]
```diff
@@ -42,6 +42,7 @@ public RetrievedDataset from(CkanPackage ckanPackage) {
                 .url(ckanPackage.getUrl())
                 .languages(values(ckanPackage.getLanguage()))
                 .contact(value(ckanPackage.getContactUri()))
+                .creator(creators(ckanPackage))
```
Comment (PatrickDekkerHealthRI) [inline src/main/java/io/github/genomicdatainfrastructure/discovery/utils/PackageShowMapper.java:45]: creator**s** is accept a list of creators.

File: src/test/java/io/github/genomicdatainfrastructure/discovery/services/PackageShowMapperTest.java [hunk 2]
```diff
@@ -46,6 +47,7 @@ void accepts_empty_package() {
                 .themes(List.of())
                 .keywords(List.of())
                 .contacts(List.of())
+                .creator(List.of())
```
Comment (PatrickDekkerHealthRI) [inline src/test/java/io/github/genomicdatainfrastructure/discovery/services/PackageShowMapperTest.java:50]: creator**s**

File: src/main/openapi/ckan.yaml [hunk 3]
```diff
@@ -224,6 +228,16 @@ components:
       required:
         - name
         - email
+    Creator:
+      type: object
+      properties:
+        creator_name:
```
Comment (PatrickDekkerHealthRI) [inline src/main/openapi/ckan.yaml:234]: i think "name" should be ok instead of "creator_name" , same for create_identifier..

**Review by PatrickDekkerHealthRI — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/genomicdatainfrastructure/discovery/utils/PackageShowMapper.java (hunk 1 above)
Comment (Markus92) [reply src/main/java/io/github/genomicdatainfrastructure/discovery/utils/PackageShowMapper.java]: Done

File: src/test/java/io/github/genomicdatainfrastructure/discovery/services/PackageShowMapperTest.java (hunk 2 above)
Comment (Markus92) [reply src/test/java/io/github/genomicdatainfrastructure/discovery/services/PackageShowMapperTest.java]: Done

File: src/main/openapi/ckan.yaml [hunk 4]
```diff
@@ -164,6 +164,10 @@ components:
           items:
             $ref: "#/components/schemas/CkanContactPoint"
           title: Contacts
+        creators:
+          type: array
+          items:
+            $ref: "#/components/schemas/Creator"
```
Comment (brunopacheco1) [inline src/main/openapi/ckan.yaml:170]: important: please add `Ckan` to class name to avoid name clashing, if we decide to support other metadata sources.

**Comment by brunopacheco1** [comment]
@Markus92 thanks for working on it.

Can we discuss on required metadata parallel? I need to align `creator` and other metadata we currently have mapped here, that possibly are not required at all.

**Review by brunopacheco1 — CHANGES_REQUESTED** [review]

File: src/main/openapi/ckan.yaml [hunk 5]
```diff
@@ -224,6 +228,16 @@ components:
...
         - name
         - email
+    Creator:
+      type: object
+      properties:
+        creator_name:
+          type: string
+        creator_identifier:
+          type: string
+      required:
```
[the remark this answers is not in the document]
Comment (Markus92) [reply src/main/openapi/ckan.yaml:238]: Agreed, but currently we only have publisher_name that we expose?

File: src/main/openapi/ckan.yaml (hunk 4 above)
Comment (Markus92) [reply src/main/openapi/ckan.yaml]: Great catch, I added the prefix in the latest revision.

File: src/main/openapi/ckan.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (zalborzi) [reply src/main/openapi/ckan.yaml:238]: That's true. This can go to a specific enhancement/refactoring ticket once we have the metadata list.

**Review by brunopacheco1 — APPROVED** [review]

**Review by hcvdwerf — APPROVED** [review]

**Review by PatrickDekkerHealthRI — APPROVED** [review]
LGTM

**Pull request merged**